lib/filewatch/read_mode/handlers/read_zip_file.rb

```diff
@@ -59,5 +68,24 @@ def close_and_ignore_ioexception(closeable)
         logger.warn("Ignoring an IOException when closing an instance of #{closeable.class.name}", "exception" => e)
       end
     end
+
+    def corrupted?(watched_file)
+      begin
+        file_stream = FileInputStream.new(watched_file.path)
+        gzip_stream = GZIPInputStream.new(file_stream)
+        buffer = Java::byte[8192].new
+        start = Time.new
+        until gzip_stream.read(buffer) == -1
+        end
+        return false
+      rescue ZipException => e
+        duration = Time.now - start
+        logger.warn("#{watched_file.path} is corrupted, #{duration} secs, message: #{e.message}")
```

Comment:
would it make sense to log using an easily identifiable prefix to be able to easily go through logs? e.g.
```suggestion
        logger.warn("Detected corrupted archive #{watched_file.path} file won't be processed", :message => e.message, :duration => duration.round(3))`
```